website/project/model.py

```diff
@@ -1411,6 +1411,35 @@ def set_node_license(self, license_id, year, copyright_holders, auth, save=False
         if save:
             self.save()
 
+    def subscribe_user_to_notifications(self, user):
+        """ Update the notification settings for the creator or contributors
+
+        :param user: User to subscribe to notifications
+        """
+        from website.notifications.utils import to_subscription_key
```

Comment:
As suggested in a previous comment, it would be good to move as much notifications-specific code out of the project model. All these circular imports smell of mixed concerns.

A common solution we've been using is signals. How about creating listeners for the `contributor_added` (already exists) and `project_created` (doesn't yet exist) signals? 

The code will look something like:

```python
# website.notifications.listeners

from website.project.signals import project_created, contributor_added

def subscribe_user_to_notifications(node, user):
    # ...

@project_created.connect
def subscribe_creator(node):
    subscribe_user_to_notifications(node, node.creator)


@contributor_added.connect
def subscribe_contributor(node, contributor, *args, **kwargs):
    subscribe_user_to_notifications(node, contributor)
```

`subscribe_user_to_notifications` will essentially replace the method you have here.